internal/provider/models/engine_datasource.go

```diff
@@ -0,0 +1,192 @@
+package models
+
+import (
+	"github.com/hashicorp/terraform-plugin-framework/datasource/schema"
+
+	"github.com/incident-io/terraform-provider-incident/internal/apischema"
+)
+
+// Data source attribute helpers (computed versions)
+
+func ParamBindingValueDataSourceAttributes() map[string]schema.Attribute {
+	return map[string]schema.Attribute{
+		"literal": schema.StringAttribute{
+			MarkdownDescription: apischema.Docstring("EngineParamBindingValueV2", "literal"),
+			Computed:            true,
+		},
+		"reference": schema.StringAttribute{
+			MarkdownDescription: apischema.Docstring("EngineParamBindingValueV2", "reference"),
+			Computed:            true,
+		},
+	}
+}
+
+func ParamBindingDataSourceAttributes() map[string]schema.Attribute {
+	return map[string]schema.Attribute{
+		"array_value": schema.SetNestedAttribute{
+			MarkdownDescription: "The array of literal or reference parameter values",
```

Comment:
I think you should be able to use `apischema.Docstring` for all of these instead of hardcoding the description. That will ensure it's inline with our API documentation.